src/main/java/com/epam/xmlparsing/entity/Plant.java

```diff
@@ -0,0 +1,143 @@
+package com.epam.xmlparsing.entity;
+
+import javax.xml.bind.annotation.*;
+import java.util.Objects;
+
+@XmlRootElement(name = "plant", namespace = "plants")
+@XmlSeeAlso({Flower.class, Tree.class})
+@XmlAccessorType(XmlAccessType.FIELD)
+public class Plant {
+
+    @XmlAttribute(required = true)
+    private int id;
+    @XmlAttribute
+    private String origin = "";
+    @XmlElement(namespace = "plants")
+    private String name;
+    @XmlElement(namespace = "plants")
+    private Soil soilType;
+    @XmlElement(name = "growingTips", namespace = "plants")
+    private GrowingTips growingTips;
+
+    public Plant() {
+    }
+
+    public Plant(int id, String origin, String name, Soil soilType, int neededTemperature, boolean needsLight, int weeklyRequiredWater) {
```

Comment:
better to inject object here instead of last 3 params to have a better control over the dependency injection, and makes dependencies of a class explicit and visible